backend/cmd/headlamp.go

```diff
@@ -719,11 +724,54 @@ func handleClusterHelm(c *HeadlampConfig, router *mux.Router) {
 	})
 }
 
-func handleClusterAPI(c *HeadlampConfig, router *mux.Router) {
+func handleClusterAPI(c *HeadlampConfig, router *mux.Router) { //nolint:funlen
 	router.PathPrefix("/clusters/{clusterName}/{api:.*}").HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var contextKey string
 		clusterName := mux.Vars(r)["clusterName"]
 
-		kContext, err := c.kubeConfigStore.GetContext(clusterName)
+		// checking if kubeConfig exists, if not check if the request headers for kubeConfig information
+		kubeConfig := r.Header.Get("KUBECONFIG")
+
+		if kubeConfig != "" && c.enableDynamicClusters {
+			// if kubeConfig is set and dynamic clusters are enabled then handle stateless cluster requests
+			key, err := c.handleStatelessReq(r, kubeConfig)
+			if err != nil {
+				log.Println("Error: handling stateless cluster request: ", err)
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			contextKey = key
+		} else {
+			contextKey = clusterName
+		}
```

Comment:
If KUBECONFIG is present in all requests made by a client then the handleStatelessReq will be called everytime irrespective of the cluster being present in the kubeConfigStore.